cmake/golden-tests.cmake

```diff
@@ -1,35 +1,64 @@
 macro(add_golden_test)
+    # Parse arguments
     set(options "")
-    set(one_value_args FILE)
-    set(multi_value_args LIB)
+    set(one_value_args TESTFILE)
+    set(multi_value_args SOURCE LIB)
     cmake_parse_arguments(GT "${options}" "${one_value_args}" "${multi_value_args}" ${ARGN})
 
-    get_filename_component(test_filename ${GT_FILE} NAME_WE)
+    # Retrieve file name
+    get_filename_component(test_filename ${GT_TESTFILE} NAME_WE)
     set(target_name ${PROJECT_NAME}_${test_filename})
 
-    add_executable(${target_name} EXCLUDE_FROM_ALL ${GT_FILE})
+    # Sanity check
+    if(NOT ("${GT_TESTFILE}" MATCHES "\.test\.cpp$"))
+        # In case we are testing one of the real threads
+        if(NOT ("${GT_TESTFILE}" MATCHES ".*Sts1CobcSw/.*Thread.cpp"))
+            message(FATAL_ERROR "Name of test file `${GT_TESTFILE}` do not end with .test.cpp")
+        endif()
+    endif()
+
+    # Create executable
+    add_executable(${target_name} EXCLUDE_FROM_ALL ${GT_TESTFILE} ${GT_SOURCE})
+    target_include_directories(${target_name} ${warning_guard} PUBLIC "${CMAKE_SOURCE_DIR}")
     set_target_properties(${target_name} PROPERTIES OUTPUT_NAME ${test_filename})
     target_link_libraries(${target_name} PUBLIC ${GT_LIB})
 
-    add_custom_command(
-        OUTPUT "${test_filename}.output"
-        COMMAND bash "${CMAKE_CURRENT_SOURCE_DIR}/Scripts/TestRunner.sh"
-                $<TARGET_FILE:${target_name}>
-        DEPENDS ${target_name} Scripts/TestRunner.sh
-    )
+
+    if("${GT_TESTFILE}" MATCHES "\.test\.cpp$")
+        add_custom_command(
+            OUTPUT "${test_filename}.output"
+            COMMAND bash "${CMAKE_CURRENT_SOURCE_DIR}/Scripts/TestRunner.sh"
+                    $<TARGET_FILE:${target_name}>
+            DEPENDS ${target_name} Scripts/TestRunner.sh
+        )
+        list(APPEND output_files "${test_filename}.output")
```

Comment:
I think this line should be deleted, because you also append the output files in line 52.

Reply:
Should be fixed by https://github.com/SpaceTeam/STS1_COBC_SW/pull/73/commits/437a6f4fe7df109ff794331fbd8b5f23f8b1feb9